### PR TITLE
Update+modularize Lodash & conform to semver

### PR DESCRIPTION
**Pull request by jameswomack**
* Use Lodash modules instead of full lodash, increasing performance of
  installing and running the code
* Conforming to semantic versioning to work better with tools as well
  as how developer think about Node modules today

The version will need to be changed to 1.0.0 and then the package will need to be published to NPM.
This can be done by running `npm version major && npm publish` after merge.

I can do this also if added to the collaborators of the project in NPM's system.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,8 +29,11 @@
     "mocha": "^2.0.1"
   },
   "scripts": {
-    "test": "./node_modules/.bin/mocha --recursive -R spec spec/"
+    "test": "mocha --recursive -R spec spec/"
```
Comment (jameswomack) [inline package.json]: Putting the relative path isn't necessary when inside an npm script string.

Comment (estliberitas) [inline package.json]: alrighty then )

**Comment by estliberitas** [comment]
Squash commits plz

**Comment by estliberitas** [comment]
Also why it would be semver-major? And not patch. Though earlier I might be wrong yep. Now, I'm aware what you are talking about.

**Comment by jameswomack** [comment]
@estliberitas Normally this *would* be a patch release. The reason I've placed a major version number is that a) there's never been a major release on this module and the semantic versioning spec [requires the first release w/ a public API defined to be 1.0.0](http://semver.org/#spec-item-5) and b) NPM's use of `^` is predicated on devs following the spec. I would say your public API is well defined as a lot of folks use your module (thanks BTW) and it's been stable for a long time :smile_cat:

**Comment by jameswomack** [comment]
The squashaning has commenced
Yes, the squashaning has commenced

**Comment by estliberitas** [comment]
Deal, man! U got me, ha. And thanks for clarification. I did not knew about public API stuff in semantic versioning spec. :smile:

**Comment by estliberitas** [comment]
Btw, I thought my module sucks compared to Plywood by Imply.

**Comment by estliberitas** [comment]
Damn, I want to land it using rebase, but okay anyways

**Comment by estliberitas** [comment]
1.0.0 is out

**Comment by jameswomack** [comment]
Awesome, thanks

**Pull request merged**